implementation/src/main/java/io/smallrye/jwt/build/impl/JwtSigningUtils.java

```diff
@@ -208,12 +208,24 @@ static void setDefaultJwtClaims(JwtClaims claims) {
         if (!claims.hasClaim(Claims.iat.name())) {
             claims.setIssuedAt(NumericDate.fromSeconds(currentTimeInSecs));
         }
-        if (!claims.hasClaim(Claims.exp.name())) {
-            claims.setExpirationTime(NumericDate.fromSeconds(currentTimeInSecs() + 300));
-        }
+        setExpiryClaim(claims);
         if (!claims.hasClaim(Claims.jti.name())) {
             claims.setGeneratedJwtId();
         }
+        if (!claims.hasClaim(Claims.iss.name())) {
+            String issuer = getConfigProperty("smallrye.jwt.new-token.issuer", String.class);
+            if (issuer != null) {
+                claims.setIssuer(issuer);
+            }
+        }
+    }
+
+    static <T> T getConfigProperty(String name, Class<T> cls) {
+        try {
+            return ConfigProvider.getConfig().getValue(name, cls);
+        } catch (NoSuchElementException ex) {
+            return null;
+        }
     }
```

Comment:
It might be cleaner to have this method take a default value, then use `getOptionalValue(name, cls).orElse(defaultValue)`. You can avoid the `try`/`catch` and also the conditional logic for setting the timeout of `300L`

Reply:
Hi Mike @MikeEdgar I've tried to optimize a bit, as indeed it is only in the case of the timeout that a non null default value is available. Please check